ENIGMAsystem/SHELL/Platforms/xlib/BSD/progdir.cpp

```diff
@@ -0,0 +1,22 @@
+#include "Platforms/General/PFmain.h"
+#include "Universal_System/estring.h"
+#include "progdir.h"
+#include <cstdlib>
+#include <unistd.h>
+#include <sys/sysctl.h>
+
+namespace enigma {
+
+void initialize_program_directory() {
```

Comment:
Why are you always wrong? Jokes, with combined seriousness aside, I'm initializing the program_directory differently depending on whether it is Linux or BSD, because doing this on these platforms requires platform-specific code.